freemocap/fmc_origin_alignment.py

```diff
@@ -0,0 +1,283 @@
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from pathlib import Path 
+from rich.progress import track 
+
+def align_skeleton_with_origin(session, mediapipe_skeleton_data_XYZ, good_clean_frame_number):
+
+    #skeleton_data_path = session.dataArrayPath/'mediaPipeSkel_3d_smoothed.npy'
+    debug = True
+    right_heel_index = 30
+    right_toe_index = 32
+    left_heel_index = 29
+    left_toe_index = 31
+
+    num_pose_joints = 33
+
+    primary_foot_indices = [left_heel_index,left_toe_index]
+    secondary_foot_index = [right_heel_index]
+    num_frames = session.numFrames
+    base_frame = good_clean_frame_number #just changing the variable names to work with the code I wrote 
+    ## functions used in getting a matrix
+    def create_vector(point1,point2): 
+        """Put two points in, make a vector [point1,point2]"""
+        vector = point2 - point1
+        return vector
+
+    def create_normal_vector(vector1,vector2): 
+        """Put two vectors in, make a normal vector"""
+        normal_vector = np.cross(vector1,vector2)
+        return normal_vector
+
+    def create_unit_vector(vector): 
+        """Take in a vector, make it a unit vector"""
+        unit_vector = vector/np.linalg.norm(vector)
+        return unit_vector
+
+    def calculate_skewed_symmetric_cross_product(cross_product_vector):
```

Comment:
is this the part you got from stack overflow?

Reply:
in general, code like this that encapsulates a lot of logic in a single call is hard to debug. Better to caluculate each row independently and then stack them to get the final output

Reply:
![image](https://user-images.githubusercontent.com/17153275/168162449-fddb641c-693d-47c3-96bf-f70840971241.png)

what I got from stack exchange was this equation for rotating some unit vector a onto unit vector b

Reply:
https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d

Reply:
yeah, right on. It does seem to do the job, so it's good enough for now :)